blink/cga.c

```diff
@@ -32,17 +33,28 @@ size_t FormatCga(u8 bgfg, char buf[11]) {
                  kCgaToAnsi[bgfg & 0x0F]);
 }
 
-void DrawCga(struct Panel *p, u8 v[25][80][2]) {
+#ifdef IUTF8
+#define CURSOR L'▂'
+#else
+#define CURSOR '_'
+#endif
+
+void DrawCga(struct Panel *p, u8 v[25][80][2], int curx, int cury) {
   char buf[11];
   unsigned y, x, n, a;
   n = MIN(25, p->bottom - p->top);
   for (y = 0; y < n; ++y) {
     a = -1;
     for (x = 0; x < 80; ++x) {
-      if (v[y][x][1] != a) {
-        AppendData(&p->lines[y], buf, FormatCga((a = v[y][x][1]), buf));
+      if (x == curx && y == cury) {
+        AppendData(&p->lines[y], buf, FormatCga(0x07, buf));
+        AppendWide(&p->lines[y], CURSOR);
```

Comment:
Thank you for your code review :)

I don't think simultaneously using routines from 'blink/cga.c' and `blink/pty.c` will work well for a number of reasons. The pty buffers don't contain character/attribute data including for the cursor, and designing them to be required to be used together adds more complexity than is needed.

I am aware of the problem that the cursor can only be displayed as occupying a space position overwriting any other character "under" it - `blink/pty.c` has the same problem, and uses the same mechanism to display a cursor (which is just the underline (`_`) character in most cases). I am not aware of any technique that allows TUI output to display two glyphs in the same location, do you know how that could be done?

Reply:
> I don't think simultaneously using routines from 'blink/cga.c' and blink/pty.c will work well for a number of reasons.

@ghaerr : but my understanding is that, at the end of the day, the escape sequences from `blink/cga.c` will still go to `blink/pty.c` anyway?

Reply:
>  but my understanding is that, at the end of the day, the escape sequences from blink/cga.c will still go to blink/pty.c anyway?

No - they operate completely independently. However, both load buffers that are rendered by the code in `blink/panel.c`.

Reply:
@tkchia: To explain this a bit further - any TUI routines, including `blink/cga.`, `blink/pty.c` and some code in `blink/blinkenlights.c` can include ANSI escape sequences; except for `blink/pty.c`, these sequences are passed through unmodified to `blink/panel.c`, where they are essentially clipped to the panel window and, if visible, sent out unmodified. However, `blink/pty.c` does the additional job of interpreting ANSI escape sequences to simulate an ANSI terminal and then stores the data in a way such that its buffers can be also rendered by `blink/panel.c` by sending final ANSI escape sequences to the real terminal (emulator).

Reply:
@ghaerr : I stand corrected.  In that case I guess what I would propose, is that `blink/cga.c` do something similar to `blink/pty.c`.  That is, if the cursor is on a character cell that is not a space, then use reverse video to display the actual character, rather than using a `▂` glyph.

Thank you!

Reply:
@tkchia: Good idea, let me see what I can do :)

Reply:
@ghaerr : by the way you need to include either `<termios.h>` or `"blink/xlat.h"` in order to get `IUTF8` defined.  Thank you!

Reply:
@tkchia: thanks, I was unaware of that. I'll include `<termios.h>`.

Reply:
@tkchia: The cursor is fixed and tested (using `vframe` FreeDOS command). Thank you!